File: app/src/main/java/com/battlelancer/seriesguide/ui/people/MovieCreditsLoader.java
package com.battlelancer.seriesguide.ui.people;

import android.content.Context;
import com.battlelancer.seriesguide.SgApp;
import com.battlelancer.seriesguide.tmdbapi.SgTmdb;
import com.uwetrottmann.androidutils.GenericSimpleLoader;
import com.uwetrottmann.tmdb2.entities.Credits;
import com.uwetrottmann.tmdb2.services.MoviesService;
import retrofit2.Response;

/**
 * Loads movie credits from TMDb.
 */
public class MovieCreditsLoader extends GenericSimpleLoader<Credits> {

    private final int tmdbId;

    public MovieCreditsLoader(Context context, int tmdbId) {
        super(context);
        this.tmdbId = tmdbId;
    }

    @Override
    public Credits loadInBackground() {
        MoviesService moviesService = SgApp.getServicesComponent(getContext()).moviesService();
        try {
            Response<Credits> response = moviesService.credits(tmdbId).execute();
            if (response.isSuccessful()) {
                return response.body();
            } else {
                SgTmdb.trackFailedRequest(getContext(), "get movie credits", response);
            }
        } catch (Exception e) {
            SgTmdb.trackFailedRequest(getContext(), "get movie credits", e);
        }

        return null;
    }
}

File: app/src/main/java/com/battlelancer/seriesguide/ui/people/MovieCreditsLoader.kt
package com.battlelancer.seriesguide.ui.people

import android.content.Context
import com.battlelancer.seriesguide.SgApp
import com.battlelancer.seriesguide.util.Errors
import com.uwetrottmann.androidutils.GenericSimpleLoader
import com.uwetrottmann.tmdb2.entities.Credits

/**
 * Loads movie credits from TMDb.
 */
class MovieCreditsLoader(context: Context, private val tmdbId: Int)
    : GenericSimpleLoader<Credits?>(context) {

    override fun loadInBackground(): Credits? {
        val moviesService = SgApp.getServicesComponent(context).moviesService()
        try {
            val response = moviesService.credits(tmdbId).execute()
            if (response.isSuccessful) {
                return response.body()
            } else {
                Errors.logAndReport("get movie credits", response)
            }
        } catch (e: Exception) {
            Errors.logAndReport("get movie credits", e)
        }

        return null
    }
}
